lib_eio_linux/eio_linux.ml

```diff
@@ -452,16 +457,16 @@ let rec schedule ({run_q; sleep_q; mem_q; uring; _} as st) : [`Exit_scheduler] =
         in
         Log.debug (fun l -> l "scheduler: %d sub / %d total, timeout %s" num_jobs st.io_jobs
                       (match timeout with None -> "inf" | Some v -> string_of_float v));
-        if timeout = None && st.io_jobs = 0 then (
-          (* Nothing further can happen at this point.
+        if timeout = None && st.io_jobs = 0 && Lf_queue.has_single_element run_q then (
```

Comment:
Why is this extra condition needed? If there are no IO jobs at all then we're not even monitoring the wakeup FD, so we must be shutting down at this point anyway.

Reply:
So I assumed `st.io_jobs` could be zero while having some outstanding tasks to run, in that case the scheduler would exit prematurely. But now I have seen that there is always at least one io job running for the eventfd mechanism.

Reply:
Sorry, I was wrong about this. We still need to run everything on the run queue even if we are shutting down. I think this is the cause of the CI failures. I've pushed a fix.

Reply:
Indeed it fixes the CI, thanks for figuring out that it was the root of the issue!